data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/AbstractLogstashPluginAttributesMapper.java

```diff
@@ -36,15 +36,24 @@
                 .stream()
                 .filter(logstashAttribute -> !customMappedAttributeNames.contains(logstashAttribute.getAttributeName()))
                 .forEach(logstashAttribute -> {
-            if (mappedAttributeNames.containsKey(logstashAttribute.getAttributeName())) {
-                pluginSettings.put(
-                        mappedAttributeNames.get(logstashAttribute.getAttributeName()),
-                        logstashAttribute.getAttributeValue().getValue()
-                );
-            }
-            else {
-                LOG.warn("Attribute name {} is not found in mapping file.", logstashAttribute.getAttributeName());
-            }
+                    final String logstashAttributeName = logstashAttribute.getAttributeName();
+
+                    if (mappedAttributeNames.containsKey(logstashAttributeName)) {
+                        if (mappedAttributeNames.get(logstashAttributeName).startsWith("!")) {
```

Comment:
I think you should extract this into a variable: 

```
String attributeValue = mappedAttributeNames.get(logstashAttributeName)
```

It is used three times, so consolidating it into a variable will aid readability.